types/index.d.ts

```diff
@@ -118,6 +118,12 @@ declare module 'mongoose' {
 
   export type HydratedDocument<DocType, TMethodsAndOverrides = {}, TVirtuals = {}> = DocType extends Document ? Require_id<DocType> : (Document<unknown, any, DocType> & Require_id<DocType> & TVirtuals & TMethodsAndOverrides);
 
+  export type HydratedDocumentFromSchema<TSchema extends Schema> = HydratedDocument<
```

Comment:
maybe consider adding a comment describing what this type is meant to be for (and how to properly use it)?